### PR TITLE
linx-server: init at unstable-2021-12-24

### LINKED ISSUE #188667: Package Request: linx-server
**Project description**
Self-hosted file/code/media sharing website.

**Metadata**

* homepage URL: https://put.icu
* source URL: https://github.com/ZizzyDizzyMC/linx-server
* license: GPLv3
* platforms: Linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #188667

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 202978` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>linx-server: log was empty</li>
  </ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
linx-server:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/servers/web-apps/linx-server/default.nix:27:5:

```
   |
27 |     license = licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

File: pkgs/servers/web-apps/linx-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    repo = pname;
+    rev = "3f503442f10fca68a3212975b23cf74d92c9988c";
+    hash = "sha256-h84s8mo3Walp/VknHI2RR0qjJLjJwfdJ0Nw2hOMIOwM=";
+  };
+
+  vendorHash = "sha256-/N3AXrPyENp3li4X86LNXsfBYbjJulk+0EAyogPNIpc=";
+
+  nativeBuildInputs = [ go-rice ];
+
+  postPatch = "rice embed-go";
```
Comment (SuperSandro2000) [inline pkgs/servers/web-apps/linx-server/default.nix:22]:
```suggestion
  preBuild = "rice embed-go";
```

**Comment by SuperSandro2000** [comment]
Tests fail due to nix store timestampts I would guess:

``` 
    server_test.go:499: File expiry is not 1669604756 but 0
--- FAIL: TestPostJSONUploadMaxExpiry (0.00s)
```

**Comment by urandom2** [comment]
that was my thought too, but it breaks with `go test` in the shell too

**Comment by SuperSandro2000** [comment]
ofborg is failing to build this packages and you posted it three times in the thread. Before posting it again please fix the build.

Disabling all tests is not great and if you don't know how to fix it we should skip this one.

**Comment by urandom2** [comment]
> ofborg is failing to build this packages and you posted it three times in the thread. Before posting it again please fix the build.

maybe I am looking at a different dashboard than you, but I see 17 successful checks and 4 neutral checks. can you point me to where you are seeing "ofborg is failing to build this packages"?

also, I am sorry for repeatedly posting this change in discourse, but I waited the requested week between each post. please let me know if I should be using that thread differently, but I was posting it there to get review feedback. 

> Disabling all tests is not great and if you don't know how to fix it we should skip this one.

as I indicated before, this appears to legitimately be a broken upstream test. is it our responsibility to fix bugs in upstream tools, or just faithfully package them as they are?
```

**Pull request merged**